### PR TITLE
feat(wrapper): add Chip component

### PR DESCRIPTION
**Pull request by duffey**
It can be used like:

```js
const container = document.createElement("div");
container.id = "spicetify-chip-examples";
container.style.width = '100%';
container.style.padding = '10px';
document.body.appendChild(container);

// JSX
ReactDOM.render(<div>

  <Spicetify.ReactComponent.Chip
    className="encore-dark-theme"
    isUsingKeyboard={false}
    onClick={(e: Event) => {
      Spicetify.showNotification("Chip clicked!");
    }}
  >Click me</Spicetify.ReactComponent.Chip>

  <Spicetify.ReactComponent.Chip
    className="encore-dark-theme my-class"
    isUsingKeyboard={false}
    size={true}
  >I'm resizeable</Spicetify.ReactComponent.Chip>

  <Spicetify.ReactComponent.Chip
    className="encore-dark-theme"
    isUsingKeyboard={false}
    iconTrailing={() => (
      <Spicetify.ReactComponent.IconComponent
        semanticColor="textBase"
        dangerouslySetInnerHTML={{ __html: Spicetify.SVGIcons["play"] }}
        iconSize={16}
        onClick={() => {
          Spicetify.showNotification("Trailing icon clicked!");

        }}
      />
    )}
  >Click my icon</Spicetify.ReactComponent.Chip>

</div>, container);

// Using `createElement`
const container2 = document.createElement("div");
container2.id = "spicetify-chip-examples-2";
container2.style.width = '100%';
container2.style.padding = '10px';
document.body.appendChild(container2);

const chip = React.createElement(Spicetify.ReactComponent.Chip, {
  className: "encore-dark-theme"
}, 'test')
ReactDOM.render(chip, container2);
```

Screenshot:

image

Please let me know if I need to modify anything else to make it easier to use or otherwise.

### CODE REVIEW

**Comment by ohitstom** [comment]
Any chance you could also provide a non-jsx code example and screenshots of potential usage.
Until docs are updated we point developers to pull requests for example usage, so it would be helpful!

**Comment by duffey** [comment]
I made some examples here: https://github.com/duffey/spicetify-chip-examples

**Comment by duffey** [comment]
I understand now you meant creating the components in pure JS, so I will try to do that.

**Comment by ohitstom** [comment]
> I understand now you meant creating the components in pure JS, so I will try to do that.

you can just modify your original pr message to include this stuff 👍

**Comment by duffey** [comment]
> > I understand now you meant creating the components in pure JS, so I will try to do that.
> 
> you can just modify your original pr message to include this stuff 👍

OK!

File: jsHelper/spicetifyWrapper.js [hunk 1]
```diff
@@ -412,6 +413,7 @@ window.Spicetify = {
 				ctaText: functionModules.find(m => m.toString().includes("ctaText")),
 				styledImage: functionModules.find(m => m.toString().includes("placeholderSrc"))
 			},
+			Chip: modules.find(m => m?.render?.toString().includes("invertedDark")),
```
Comment (rxri) [inline jsHelper/spicetifyWrapper.js:416]:
```suggestion
			Chip: modules.find(m => m?.render?.toString().includes("invertedDark") && m?.render?.toString().includes("isUsingKeyboard")),
```

**Review by rxri — CHANGES_REQUESTED** [review]

**Review by rxri — APPROVED** [review]

**Comment by afonsojramos** [comment]
Oh nice! Thanks @duffey

**Review by afonsojramos — APPROVED** [review]

**Pull request merged**